pkg/ansible/generate.go

```diff
@@ -0,0 +1,257 @@
+# vpc + subnet.
+#openshift_aws_subnet_name: cluster-engine-subnet-1
+
+# -------------- #
+# Security Group #
+# -------------- #
+
+# openshift_aws_create_security_groups defaults to true.  If you wish to use
+# an existing security group, set this to false.
+openshift_aws_create_security_groups: true
+
+# openshift_aws_build_ami_group is the name of the security group to build the
+# ami in.  This defaults to the value of openshift_aws_clusterid.
+#openshift_aws_build_ami_group: cluster-engine
+
+# openshift_aws_launch_config_security_groups specifies the security groups to
+# apply to the launch config.  The launch config security groups will be what
+# the cluster actually is deployed in.
+openshift_aws_launch_config_security_groups:
+  compute:
+  - cluster-engine
+  - cluster-engine_compute
+  - cluster-engine_compute_k8s
+  infra:
+  - cluster-engine
+  - cluster-engine_infra
+  - cluster-engine_infra_k8s
+  master:
+  - cluster-engine
+  - cluster-engine_master
+  - cluster-engine_master_k8s
+
+# openshift_aws_node_security_groups are created when
+# openshift_aws_create_security_groups is set to true.
+#openshift_aws_node_security_groups: see roles/openshift_aws/defaults/main.yml
+
+# -------- #
+# ssh keys #
+# -------- #
+
+# Specify the key pair name here to connect to the provisioned instances.  This
+# can be an existing key, or it can be one of the keys specified in
+# openshift_aws_users
+openshift_aws_ssh_key_name: libra # myuser_key
+
+# This will ensure these user and public keys are created.
+#openshift_aws_users:
+#- key_name: myuser_key
+#  username: myuser
+#  pub_key: |
+#         ssh-rsa AAAA
+
+# When building the AMI, specify the user to ssh to the instance as.
+# openshift_aws_build_ami_ssh_user: root
+
+# --------- #
+# AMI Build #
+# --------- #
+# Variables in this section apply to building a node AMI for use in your
+# openshift cluster.
+
+# must specify a base_ami when building an AMI
+openshift_aws_base_ami: ami-ae7bfdb8 # ami-12345678
+
+# this is the ami we built
+openshift_aws_ami: ami-9a0d95e0 # ami-2f53ce55
+
+# when creating an encrypted AMI please specify use_encryption
+#openshift_aws_ami_encrypt: False
+
+# -- #
+# S3 #
+# -- #
+
+# Create an s3 bucket.
+openshift_aws_create_s3: false
+
+# --- #
+# ELB #
+# --- #
+
+# openshift_aws_elb_name will be the base-name of the ELBs.
+# TODO: looks like this is supposed to just be basename variant
+#openshift_aws_elb_name: dgoodwin-dev
+
+# custom certificates are required for the ELB
+openshift_aws_iam_cert_path: /home/abutcher/rhat/certs/wildcard-flibberty-jibbet.com.crt  # '/path/to/wildcard.<clusterid>.example.com.crt'
+openshift_aws_iam_cert_key_path: /home/abutcher/rhat/certs/wildcard-flibberty-jibbet.com.key  # '/path/to/wildcard.<clusterid>.example.com.key'
+openshift_aws_iam_cert_chain_path: /home/abutcher/rhat/certs/ca.crt  # '/path/to/cert.ca.crt'
+
+openshift_aws_create_iam_role: True
+
+{{ range $key, $value := . }}
```

Comment:
Perhaps this is just an over-simplified example start, but using a template here seems to me to be a convoluted way of adding more variables at the end of a string. Do you anticipate that there will be actual variable replacement elsewhere in the template at some point?

Reply:
We're about to get to the more complicated stuff where we need a bunch of things per node group defined, some of them won't be simple string to string maps either. In all cases though they will still be appending items to the end of a string, I'm not sure which will play out easier though, the template looping might be handy when we get to:

```
openshift_aws_launch_config_security_groups:
  compute:
  - cluster-engine
  - cluster-engine_compute
  - cluster-engine_compute_k8s
  infra:
  - cluster-engine
  - cluster-engine_infra
  - cluster-engine_infra_k8s
  master:
  - cluster-engine
  - cluster-engine_master
- cluster-engine_master_k8s
```